src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js

```diff
@@ -1,1058 +1,611 @@
-                var activeFields = [];
-                _allFields.forEach(function(currentField) {
-                    if (currentField.isActive) {
-                        activeFields.push(currentField);
-                    }
-                });
-                return _sortFieldsByOrder(activeFields);
-            }
+    /**
+     * @description Checks for required fields name and description
+     * @return Boolean if user can proceed to next step
+     */
+    checkBatchInfoValidity: function (component) {
+        let batchInfo = component.get('v.batchInfo');
+        let isValid = batchInfo.name && batchInfo.description;
+        if (isValid) {
+            this.clearError(component);
+        } else {
+            component.set('v.wizardMetadata.errorMessage', component.get('v.wizardMetadata.labels.missingNameDescriptionError'));
+        }
+        return isValid;
+    },
 
-            /**
-             * @description Gets the active fields grouped by SObject.
-             * @return Map of SObject group to List of related active fields.
-             */
-            function getActivesBySObject() {
-                var activeFields = [];
-                _allFields.forEach(function(currentField) {
-                    if (currentField.isActive) {
-                        activeFields.push(currentField);
-                    }
-                });
-                activeFields = _sortFieldsByOrder(activeFields);
-                return _groupFieldsBySObject(activeFields);
-            }
+    /**
+     * @description Checks validity object on every lightning:input field
+     * @return Boolean if user can proceed to next step
+     */
+    checkBatchFieldOptionsValidity: function (component) {
+        var isValid = component.find('defaultValueField').reduce(function (validSoFar, defaultValueField) {
+            return validSoFar && defaultValueField.get('v.validity').valid;
+        }, true);
+        if (isValid) {
+            this.clearError(component);
+        }
+        return isValid;
+    },
 
-            /**
-             * @description Gets the aggregate validity of provided default values
-             * @return Boolean
-             */
-            function getDefaultFieldValidity(component) {
-                var isValid = component.find("defaultValueField").reduce(function (validSoFar, defaultValueField) {
-                    return validSoFar && defaultValueField.get("v.validity").valid;
-                }, true);
-                return isValid;
-            }
+    /**
+     * @description Checks for required fields Donation Date Range and Batch Process Size
+     * @return Boolean if user can proceed to next step
+     */
+    checkBatchProcessingSettingsValidity: function (component) {
+        const batchInfo = component.get('v.batchInfo');
+        let isValid = true;
+        let errormsg = component.get('v.wizardMetadata.labels.missingFieldsError');
+
+        if (batchInfo.donationDateRange === '') {
```

Comment:
Would we ever need to check for `null` or `undefined` here?

Reply:
(Same question below)

Reply:
Defaults are provided, and if the user deletes the value, it passes back as an empty string, so no, I don't believe we need to check further for `null` or `undefined`.